modules/webrequest-pipeline/sources/page-store.es

```diff
@@ -136,9 +136,17 @@ export default class PageStore {
   }
 
   onBeforeNavigate = (details) => {
-    const { frameId, tabId, url } = details;
+    const { frameId, tabId, url, timeStamp } = details;
     const tabContext = this.tabs.get(tabId);
     if (frameId === 0) {
+      if (
+        tabContext
+        && tabContext.id === tabId
+        && tabContext.url === url
+        && tabContext.created + 200 > timeStamp
+      ) {
+        return;
```

Comment:
Since this is a workaround for a browser bug, which hopefully should get fixed eventually, maybe we should log a warning here?

Without the context, it is not obvious why this check exists.

Reply:
Hopefully this codebase wont live for way longer. I've added to comments to explain the context